x-pack/plugins/aiops/public/shared_components/pattern_analysis.tsx

```diff
@@ -0,0 +1,172 @@
+  coreStart: CoreStart;
+  pluginStart: AiopsPluginStartDeps;
+};
+
+export type PatternAnalysisSharedComponent = FC<PatternAnalysisProps>;
+
+export interface PatternAnalysisProps {
+  dataViewId: string;
+  timeRange: TimeRange;
+  fieldName: string | undefined;
+  minimumTimeRangeOption: MinimumTimeRangeOption;
+  randomSamplerMode: RandomSamplerOption;
+  randomSamplerProbability: RandomSamplerProbability;
+  /**
+   * Component to render if there are no patterns found
+   */
+  emptyState?: React.ReactElement;
+  /**
+   * Outputs the most recent patterns data
+   */
+  onChange?: (patterns: Category[]) => void;
+  /**
+   * Last reload request time, can be used for manual reload
+   */
+  lastReloadRequestTime?: number;
+  /** Origin of the embeddable instance */
+  embeddingOrigin?: string;
+  onLoading: (isLoading: boolean) => void;
+  onRenderComplete: () => void;
+  onError: (error: Error) => void;
+}
+
+const PatternAnalysisWrapper: FC<PatternAnalysisPropsWithDeps> = ({
+  // Component dependencies
+  coreStart,
+  pluginStart,
+  // Component props
+  dataViewId,
+  timeRange,
+  fieldName,
+  minimumTimeRangeOption,
+  randomSamplerMode,
+  randomSamplerProbability,
+  onLoading,
+  onError,
+  onRenderComplete,
+  embeddingOrigin,
+  lastReloadRequestTime,
+  onChange,
+}) => {
+  const deps = useMemo(() => {
+    const { http, uiSettings, notifications, ...startServices } = coreStart;
+    const { lens, data, usageCollection, fieldFormats, charts } = pluginStart;
+
+    return {
+      http,
+      uiSettings,
+      data,
+      notifications,
+      lens,
+      usageCollection,
+      fieldFormats,
+      charts,
+      ...startServices,
+    };
+  }, [coreStart, pluginStart]);
+
+  const datePickerDeps = {
+    ...pick(deps, ['data', 'http', 'notifications', 'theme', 'uiSettings', 'i18n']),
+    uiSettingsKeys: UI_SETTINGS,
+  };
+
+  const aiopsAppContextValue = useMemo<AiopsAppDependencies>(() => {
+    return {
+      embeddingOrigin: embeddingOrigin ?? EMBEDDABLE_ORIGIN,
+      ...deps,
+    } as unknown as AiopsAppDependencies;
+  }, [deps, embeddingOrigin]);
+
+  const [manualReload$] = useState<BehaviorSubject<number>>(
+    new BehaviorSubject<number>(lastReloadRequestTime ?? Date.now())
+  );
+
+  useEffect(
+    function updateManualReloadSubject() {
+      if (!lastReloadRequestTime) return;
+      manualReload$.next(lastReloadRequestTime);
+    },
+    [lastReloadRequestTime, manualReload$]
+  );
+
+  const resultObservable$ = useMemo<Observable<number>>(() => {
+    return combineLatest([manualReload$]).pipe(
+      map(([manualReload]) => Math.max(manualReload)),
+      distinctUntilChanged()
+    );
+  }, [manualReload$]);
+
+  // TODO: Remove data-shared-item as part of https://github.com/elastic/kibana/issues/179376>
+  return (
+    <div
+      data-shared-item=""
+      data-test-subj="aiopsEmbeddablePatternAnalysis"
+      css={{
+        width: '100%',
+        padding: '10px',
```

Comment:
nit, can we use eui padding instead?

Reply:
I cannot find an appropriate eui size for `10px` and I have been asked in a comment above to use `10px` to match the padding of other embeddables.